docgen/src/guides/calendar-widget.md

```diff
@@ -0,0 +1,198 @@
+---
+title: Create a calendar widget
+mainTitle: Guides
+layout: main.pug
+name: customize
+category: guides
+withHeadings: true
+navWeight: 20
+editable: true
+githubSource: docgen/src/guides/calendar-widget.md
+---
+
+Search results often need to be refined by dates. Calendars are a good visual way to improve the user experience when it comes to filtering dates. InstantSearch provides custom widgets to create reusable components within your app.
+
+By the end of this guide, you'll understand how to implement a calendar widget to refine your results based on a single date and a range of dates. You can [preview](https://community.algolia.com/instantsearch.js/v2/examples/calendar-widget/) and [download](https://community.algolia.com/instantsearch.js/v2/examples/calendar-widget.zip) the concert app that you'll be able to build.
+
+## Before we start
+
+Algolia handles dates as numeric values. We'll store them as JavaScript timestamps, which are the number of milliseconds since the year 1970 ([see `Date.prototype.getTime()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTime)).
+
+Through this guide, we'll use a generated concert dataset that contains the following attributes:
+
+* `name`: the name of the music band
+* `location`: the venue of the concert
+* `date`: the date of the event we will refine the results with
+
+Both `name` and `location` will be added to the [searchable attributes](https://www.algolia.com/doc/guides/ranking/searchable-attributes/) in the Algolia dashboard so that the user can filter events on their name *and* their location.
+
+To learn more about creating your own dataset and uploading it to an Algolia index, you can read the guide on [importing from the dashboard](https://www.algolia.com/doc/tutorials/indexing/importing-data/importing-from-the-dashboard/).
+
+## Bootstrapping an InstantSearch app
+
+If you don't yet have an InstantSearch app to work on, you can use the command-line tool [`create-instantsearch-app`](https://github.com/algolia/create-instantsearch-app) that bootstraps an app for you. It will provide a basic JavaScript application including [instantsearch.js](https://github.com/algolia/instantsearch.js) itself. Make sure to create an index in your Algolia dashboard before any further steps.
+
+If you're using [npm](https://www.npmjs.com/):
+
+```console
+npm install -g create-instantsearch-app
+create-instantsearch-app [name-of-your-new-app]
+```
+
+If you're using [Yarn](https://yarnpkg.com):
+
+```console
+yarn create instantsearch-app [name-of-your-new-app]
+```
+
+The command-line tool will ask for your credentials:
+
+![create-instantsearch-app](images/create-instantsearch-app-events.png)
+
+## Customizing the results
+
+The default `item` template from the `hits` widget provided by `create-instantsearch-app` is great, but we're going to enhance it to display the relevant information of our dataset.
+
+```javascript
+search.addWidget(
```

Comment:
this is `runnable` for some reason, but it doesn't work with the demo `instant_search` index; @bobylito can we make that configurable in the guide metadata 🤔 ?

Reply:
Oh :/ It's because we  do:

```js
    if (exampleContent.indexOf("search.addWidget") === 0) {
      initWidgetExample(codeSample, index);
    }
``` 

in the runnable code JS :/

Reply:
It would be nice if the index would be configurable based on the page metadata, do you think that's possible @bobylito ?

Reply:
Working on it!

Reply:
PR sent #2690